src/services/SQSService.js

```diff
@@ -5,21 +5,35 @@ import sgMail from '@sendgrid/mail';
 import logger from '../utils/logger';
 import NotificationTypeModel from '../models/notificationType';
 import NotificationTemplateModel from '../models/notificationTemplate';
+import EmailMessageLog from '../models/emailMessageLog';
 
 const sqs = new AWS.SQS({ region: process.env.USERPOOL_REGION });
 const QUEUE_URL = `https://sqs.${process.env.USERPOOL_REGION}.amazonaws.com/${process.env.ACCOUNT_ID}/${process.env.DELIVER_EMAIL_QUEUE_NAME}`;
 
 class SQSService {
-  deliverEmail = (event, context, callback) => {
+  deliverEmail = async (event, context, callback) => {
     context.callbackWaitsForEmptyEventLoop = false;
 
     try {
+      const now = new Date();
+      const emailLog = await EmailMessageLog.findById(JSON.parse(event.Records[0].body).emailLogId).exec();
+      emailLog.set({
+        initialRequestType: JSON.parse(event.Records[0].body).type,
```

Comment:
We should not be updating any of the originalRequest or generateEmailMessage fields here because they are inteded to track information from different processes. We should only be setting the deliverEmailMessageDate field to the current date/time and the deliverEmailMessage field to the object that we will be passing to the sgMail.send() method.

Reply:
done